File: aoc2017/day4.swift
//
//  day4.swift
//  aoc2017
//
//  Created by Chris McElroy on 7/23/21.
//

import Foundation

func day4() {
	var validCount = 0
	
	for line in input {
		var valid = true
		var words: Set<Set<Character>> = []
		for word in line.split(separator: " ") {
			if !words.insert(Set(word)).inserted {
				valid = false
				break
			}
		}
		if valid { validCount += 1 }
	}
	
	print(validCount)
}
// 451 223

File: go/day4.go
package main

func day4() {
	validCount := 0

	for _, line := range wordLines {
		words := stringSet{map[string]struct{}{}}
		for i, word := range line {
			if !words.insert(SortString(word)) {
				break
			} else if i == len(line)-1 {
				validCount += 1
			}
		}
	}

	println(validCount)
}
